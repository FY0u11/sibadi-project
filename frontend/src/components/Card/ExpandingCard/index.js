import React from 'react'

import './styles.css'

const ExpandingCard = ({ image, title, body, exBody, isExpanded, width = '500px' }) => {
  return (
    <div className="my-expanding-card" style={{ width }}>
      <div className={`card ${isExpanded && 'my-expanded'}`}>
        <div className="row no-gutters">
          <div className="col-md-4">{image}</div>
          <div className="col-md-8">
            <div className="card-body">
              <h5 className="card-title">{title}</h5>
              {body}
            </div>
          </div>
        </div>
      </div>
      {isExpanded && <div className="my-expanded-body">{exBody}</div>}
    </div>
  )
}

export default ExpandingCard
