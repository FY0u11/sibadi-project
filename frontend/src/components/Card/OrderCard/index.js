import React, { useState, useEffect } from 'react'
import { connect } from 'react-redux'

import { updateOrderRequest } from '../../../utils/api'
import { setUser } from '../../../store/actions'
import { updateOrder } from '../../../store/scenes/orders/actions'
import { setModal, closeModal } from '../../../store/components/modal/actions'
import { setNotificationError } from '../../../store/components/notifications/actions'

import Button from '../../Button'
import OrderCard from './OrderCard'

const OrderCardContainer = ({
  user,
  setUser,
  showCancelWhen,
  body,
  isExpanded,
  order,
  updateOrder,
  setModal,
  closeModal,
  products,
  setNotificationError
}) => {
  const [orderProducts, setOrderProducts] = useState([])
  useEffect(() => {
    const orderProductsTmp = []
    order.orderProducts.forEach(orderProductId => {
      const product = products.find(product => product.id === orderProductId)
      orderProductsTmp.push(product)
    })
    setOrderProducts(orderProductsTmp)
  }, [])

  const cancelOrder = async () => {
    try {
      await updateOrderRequest({ status: 'cancelled' }, order.id)
      updateOrder(order.id, { status: 'cancelled' })
      setUser({ ...user, budget: user.budget + +order.price })
      closeModal()
    } catch (error) {
      setNotificationError(error.message)
    }
  }

  const askForCancellingOrder = () => {
    setModal({
      title: 'This cancels the order',
      body: (
        <div style={{ padding: '10px' }}>
          <div style={{ marginBottom: '20px' }}>Are you sure?</div>
          <Button action={cancelOrder}>Yes</Button>
          <Button action={closeModal}>No</Button>
        </div>
      )
    })
  }

  return (
    <OrderCard
      isCancelButtonShown={showCancelWhen.includes(order.status)}
      body={body}
      isExpanded={isExpanded}
      orderProducts={orderProducts}
      order={order}
      askForCancellingOrder={askForCancellingOrder}
    />
  )
}

const putStateToProps = state => ({
  user: state.root.user,
  products: state.products.products
})

const putActionsToProps = { setUser, updateOrder, setModal, closeModal, setNotificationError }

export default connect(putStateToProps, putActionsToProps)(OrderCardContainer)
