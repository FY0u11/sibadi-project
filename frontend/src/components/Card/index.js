import React from 'react'

const Card = ({ image, title, body, width = '500px' }) => {
  return (
    <div className="card mb-3 my-card" style={{ margin: '0 auto', width }}>
      <div className="row no-gutters">
        <div className="col-md-4">{image}</div>
        <div className="col-md-8">
          <div className="card-body">
            <h5 className="card-title">{title}</h5>
            {body}
          </div>
        </div>
      </div>
    </div>
  )
}

export default Card
