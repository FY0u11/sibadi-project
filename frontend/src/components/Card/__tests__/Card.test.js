import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Card from '../'

afterEach(cleanup)

describe('Testing Card component', () => {
  it('It should be properly rendered', () => {
    const props = {
      image: 'test image',
      title: 'test title',
      body: 'test body'
    }
    const { asFragment } = render(<Card {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
