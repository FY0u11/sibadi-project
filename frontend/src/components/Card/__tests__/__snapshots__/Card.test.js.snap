// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card component It should be properly rendered 1`] = `
<DocumentFragment>
  <div
    class="card mb-3 my-card"
    style="margin: 0px auto; width: 500px;"
  >
    <div
      class="row no-gutters"
    >
      <div
        class="col-md-4"
      >
        test image
      </div>
      <div
        class="col-md-8"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            test title
          </h5>
          test body
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
